src/lib/component/VariableVisibilityInput.svelte

```diff
@@ -0,0 +1,19 @@
+<script lang="ts">
+  import Icon from './Icon.svelte';
+  import Icons from './Icons';
+
+  let visible = false;
+  $: type = visible ? 'text' : 'password';
+  $: icon = visible ? Icons.Hidden : Icons.Visible;
+</script>
+
+<div class="grid">
+  <input {type} {...$$restProps} class="col-start-1 col-end-3 row-start-1 {$$props.class}" />
```

Comment:
I didn't know restProps, what it the goal here? I searched a little bit and it doesn't seem to be recommended https://github.com/sveltejs/svelte/issues/8249

Reply:
My goal here was to make the component reusable in different scenarios, but maintain the same icon placement. For that I had to have these specific classes, but still wanted to allow for more customization, hence the `$$props.class`. The `$$restProps` was to collect all the other attributes like names, placeholders, values etc. I decided to use this because inputs have lots of possible attributes, but I guess I could try to make the most common ones into optional properties maybe?

Reply:
After searching a bit more about restProps, I believe this implementation is appropriate. Svelte has since updates the docs on this:
```
$$props references all props that are passed to a component, including ones that are not declared with export. Using $$props will not perform as well as references to a specific prop because changes to any prop will cause Svelte to recheck all usages of $$props. But it can be useful in some cases – for example, when you don't know at compile time what props might be passed to a component.

<Widget {...$$props} />

$$restProps contains only the props which are not declared with export. It can be used to pass down other unknown attributes to an element in a component. It shares the same performance characteristics compared to specific property access as $$props.

<input {...$$restProps} />
```

This means that specific prop access could be slower (such as $$props.class). I found an alternative but I don't find it visually appealing. Let me know what you think:

```js
export {_class as class};
let _class = "";

<input {type} {...$$restProps} class="{_class}" />
```